app/src/main/java/com/nextcloud/talk/activities/CallActivity.java

```diff
@@ -371,6 +390,41 @@ public void onCreate(Bundle savedInstanceState) {
             setCallState(CallStatus.CONNECTING);
         }
 
+        callRecordingViewModel = new ViewModelProvider(this, viewModelFactory).get((CallRecordingViewModel.class));
+        callRecordingViewModel.setData(roomToken);
+        callRecordingViewModel.setRecordingState(extras.getInt(KEY_RECORDING_STATE));
+
+        callRecordingViewModel.getViewState().observe(this, viewState -> {
+            if (viewState instanceof CallRecordingViewModel.RecordingStartedState) {
+                binding.callRecordingIndicator.setVisibility(View.VISIBLE);
+                if (((CallRecordingViewModel.RecordingStartedState) viewState).getShowStartedInfo()) {
```

Comment:
You may want to call back to the ViewModel so that `showStartedInfo` is set back to `false`. Otherwise the state will stay and if, for example, the phone is rotated, or the user minimizes and then resumes the app, the toast and vibration will happen again.

Reply:
well ViewModel will survive configuration changes right?!
and the Observer is only triggered when the state is set which only happens for defined actions..

> for example, the phone is rotated, or the user minimizes and then resumes the app, the toast and vibration will happen again.

so could not reproduce this and i would say it's okay to keep it like this.

Reply:
Hm, may depend on how the Controllers work/are setup then. In a Fragment, the ViewModel would survive configuration changes but the Fragment would not, which means the Fragment would be recreated, and then process the state again, showing the toast again.